R/layer-sf.R

```diff
@@ -62,3 +77,9 @@ is_sf <- function(data) {
 #' @export
 scale_type.sfc <- function(x) "identity"
 
+# helper function to determine the geometry type of sf object
+sf_geometry_type <- function(sf) {
+  geometry_type <- unique(as.character(sf::st_geometry_type(sf)))
+  if (length(geometry_type) != 1)  geometry_type <- "GEOMETRY"
+  geometry_type
```

Comment:
I think this function should return `sf_types[geometry_type]` here, to simplify the code further up. I would also propose renaming it, since `sf_geometry_type()` sounds so similar to `sf::st_geometry_type()` but does somewhat different things. I recommend the name `detect_geometry_type()`.